docs/csharp/programming-guide/classes-and-structs/static-constructors.md

```diff
@@ -11,27 +11,55 @@ ms.assetid: 151ec95e-3c4d-4ed7-885d-95b7a3be2e7d
-  
-- The user has no control on when the static constructor is executed in the program.  
+- A static constructor is called automatically to initialize the [class](../../../csharp/language-reference/keywords/class.md) before the first instance is created or any static members are referenced. A static constructor will run before an instance constructor. Note that a type's static constructor is called when a static method assigned to an event or a delegate is invoked and not when it is assigned. If static field variable initializers are present in the class of the static constructor, they will be executed in the textual order in which they appear in the class declaration immediately prior to the execution of the static constructor.
+
+- If you don't provide a static constructor to initialize static fields, all static fields are initialized to their default value as listed in the [Default Values Table](../../../csharp/language-reference/keywords/default-values-table.md). 
   
+- If a static constructor throws an exception, the runtime will not invoke it a second time, and the type will remain uninitialized for the lifetime of the application domain in which your program is running. Most commonly, a <xref:System.TypeInitializationException> exception is thrown when a static constructor is unable to instantiate a type or for an unhandled exception occuring within a static constructor. For implicit static constructors that are not explicitly defined in source code, troubleshooting may require inspection of the intermediate language (IL) code.
+
+- The presence of a static constructor prevents the addition of the the <xref:System.Reflection.TypeAttributes.BeforeFieldInit> type attribute. This limits runtime optimization.
+
+- A field declared as `static readonly` may only be assigned as part of its declaration or in a static constructor. When an explicit static constructor is not required, initialize static fields at declaration, rather than through a static constructor for better runtime optimization.
+
+> [!Note]
+> Though not directly accessible, the presence of an explicit static constructor should be documented to assist with troubleshooting initialization exceptions.
+
+### Usage
+
 - A typical use of static constructors is when the class is using a log file and the constructor is used to write entries to this file.  
-  
 - Static constructors are also useful when creating wrapper classes for unmanaged code, when the constructor can call the `LoadLibrary` method.  
-  
-- If a static constructor throws an exception, the runtime will not invoke it a second time, and the type will remain uninitialized for the lifetime of the application domain in which your program is running.  
-  
-## Example  
+
+- Static constructors are also a convenient place to enforce run-time checks on the type parameter that cannot be checked at compile-time via constraints (Type parameter constraints).
+
+### Explicit initialization
+
+> [!CAUTION]
```

Comment:
I think that leaving this section out would be better. Readers that already know about `RunClassConstructor` would (hopefully) know they should not call the method. Other readers can remain blissfully unaware of its existence.

What were your thoughts regarding adding this?

Reply:
The `RunClassConstructor` was mainly added for completion's sake . It has been removed as its presence would promote misuse.

It may be worthwhile to include an example that an empty static void method would serve the same purpose.

```csharp
using System; 

public class B 
{      
    public static void TriggerStaticConstructor() {} 
	
    static B(){
        Console.WriteLine("Static constructor invoked.");
    }
} 
public class A 
{
    public static void Main(string[] args){
        B.TriggerStaticConstructor();
    }
} 
```

Reply:
Thanks for the update. I think this addition isn't necessary, so I'll :shipit: now.